.circleci/scripts/binary_windows_build.sh

```diff
@@ -5,7 +5,7 @@ source "/c/w/env"
 mkdir -p "$PYTORCH_FINAL_PACKAGE_DIR"
 
 export CUDA_VERSION="${DESIRED_CUDA/cu/}"
-export VC_YEAR=2017
+export VC_YEAR=2019
```

Comment:
Wait, before merging this PR, some changes are needed at the builder side. Otherwise, the conda packages will be broken.

Reply:
I think it's better to do that in a separate PR. Let me do this for you.

Reply:
@malfet The required changes for builder is in https://github.com/pytorch/builder/pull/445. And here I guess you'll need to make a bit modification.
```suggestion
if [[ "$CUDA_VERSION" == "92" || "$CUDA_VERSION" == "100" ]]; then
  export VC_YEAR=2017
else
  export VC_YEAR=2019
fi
```

Reply:
Actually, we can't make this change, as by convention all Python extensions on Windows must be built with VS2017:
https://docs.conda.io/projects/conda-build/en/latest/user-guide/tutorials/building-conda-packages.html

Reply:
@malfet Well, you can. But you'll need to write a custom `conda_build_config.yaml`. I've already done this for torchaudio and torchtext.

Reply:
Do VS2017 and VS2019 have binary compatible C++ ABI? If not it might cause subtle runtime failures, even if builds and smoke testing succeeds.

Reply:
See https://devblogs.microsoft.com/cppblog/cpp-binary-compatibility-and-pain-free-upgrades-to-visual-studio-2019/. It's safe to link an executable with libraries generated by VC 14.11 / 14.16  and 14.24. But it won't work if you use /GL (whole program optimization).